packages/vite/src/node/http.ts

```diff
@@ -95,23 +95,27 @@ export async function resolveHttpServer(
   httpsOptions?: HttpsServerOptions
 ): Promise<HttpServer> {
   if (!httpsOptions) {
-    return require('http').createServer(app)
+    const { createServer } = await import('http')
+    return createServer(app)
   }
 
+  // #484 fallback to http1 when proxy is needed.
   if (proxy) {
-    // #484 fallback to http1 when proxy is needed.
-    return require('https').createServer(httpsOptions, app)
+    const { createServer } = await import('http')
```

Comment:
https?

Reply:
@natrim good catch! Would you like to send a PR to fix this?

Reply:
sure, #8143